### ISSUE TITLE
magento/magento2-base module dependency issue

### ISSUE THREAD

**Issue by t-heuser** [magento/magento2#34562]
### Preconditions (*)

1. Magento 2.4.3-p1
2. Use [roave/security-advisories](https://github.com/Roave/SecurityAdvisories) to check for vulnerabilities in your dependencies

### Steps to reproduce (*)

1. Run `composer require --dev roave/security-advisories:dev-latest` or `composer update --dry-run roave/security-advisories` (if package already installed)

### Expected result (*)

1. The process should be successful.

### Actual result (*)

```console
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - magento/product-community-edition is locked to version 2.4.3-p1 and an update of this package was not requested.
    - roave/security-advisories dev-latest conflicts with tinymce/tinymce <5.10 (magento/magento2-base 2.4.3-p1 replaces tinymce/tinymce 3.4.7).
    - magento/product-community-edition 2.4.3-p1 requires magento/magento2-base 2.4.3-p1 -> satisfiable by magento/magento2-base[2.4.3-p1].
    - Root composer.json requires roave/security-advisories dev-latest -> satisfiable by roave/security-advisories[dev-latest].


Installation failed, reverting ./composer.json and ./composer.lock to their original content.

```

### Explanation
2 days ago, "tinymce/tinymce" was added as vulnerable below version 5.10 ([commit](https://github.com/Roave/SecurityAdvisories/commit/bad3752fd78f4a07acb24e56fec0366aa711f150#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34R320)). Since then it's not possible to run the security check as it always fails as in the composer.json of the  package `magento/magento2-base` you can find the following line:
```JSON
    "replace": {
        "trentrichardson/jquery-timepicker-addon": "1.4.3",
        "components/jquery": "1.11.0",
        "blueimp/jquery-file-upload": "5.6.14",
        "components/jqueryui": "1.10.4",
        "twbs/bootstrap": "3.1.0",
        "tinymce/tinymce": "3.4.7"
    },
```

Magento doesn't require tinymce directly as composer package but replaces it which is enough to trigger the security alert. Maybe the replace should be removed as I don't really see a sense in it.
---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [x] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Hotel** [comment]
Hello @oneserv-heuser,

Thanks for posting this issue!

The issue is reproducible for us and I agree with your point. So we are confirming this issue.

Thanks

**Comment by fredden** [comment]
For reference, there are several security issues listed here affecting version 3.4.7 of tinymce: https://snyk.io/vuln/npm:tinymce

**Comment by Lex50** [comment]
Pardon my ignorance, but what might be a good work-around for this?

I can remove `roave/security-advisories` with..

```
composer remove --no-update roave/security-advisories
composer update roave/security-advisories
```

but can anyone recommend a simple patch/fix to remove the conflict with `tinymce/tinymce`, that doesn't require waiting for Adobe?

**Comment by oneserv-heuser** [comment]
@Lex50 I don't think that there i a simple solution to this other than removing `roave/security-advisories` or downgrade it to a version where there is no conflict with `tinymce/tinymce` (which would make the whole point of `roave/security-advisories` useless).
If you're running `roave/security-advisories` manually you can ignore the error, but in CI/CD, no chance I think.

**Comment by Lex50** [comment]
@oneserv-heuser thank you :)

**Comment by Ocramius** [comment]
> downgrade it to a version where there is no conflict with `tinymce/tinymce`

There are only `dev-latest` and `dev-master` of `roave/security-advisories`: they point to the same ref, and, by design, they include `tinymce/tinymce` as exclusions.

The only workaround here is to say that the package replaces `"tinymce/tinymce": "x.y.z"`, where `x.y.z` is a version not affected by security issues (needs upgrading of tinymce too)

**Comment by oneserv-heuser** [comment]
> More than 6 weeks are needed for the issue logistics of a security fix. This is about one line of code.
> 
> Better rename more things and replace logos. Much more important.

More than 8 months now 😂

**Comment by mautz-et-tong** [comment]
This official repository is getting a clown show more and more. Can't wait for the community to take over and start the fork. Adobe seems to be not able to handle this

**Comment by upbott-gulshan** [comment]
Any update on this? This seems blocking upgrade to latest versions.

**Comment by renttek** [comment]
"High Priority Backlog" does not seem that highly prioritized

**Comment by Maksold** [comment]
The same problem is on 2.4.2-p2

```
composer require --dev roave/security-advisories:dev-latest                                                                                                                                                                                                                                                                                                                              

  Problem 1
    - magento/product-community-edition is locked to version 2.4.2-p2 and an update of this package was not requested.
    - roave/security-advisories dev-latest conflicts with tinymce/tinymce <5.10.7|>=6,<6.3.1 (magento/magento2-base 2.4.2-p2 replaces tinymce/tinymce 3.4.7).
    - magento/product-community-edition 2.4.2-p2 requires magento/magento2-base 2.4.2-p2 -> satisfiable by magento/magento2-base[2.4.2-p2].
    - Root composer.json requires roave/security-advisories dev-latest -> satisfiable by roave/security-advisories[dev-latest].
```

**Comment by renttek** [comment]
Any update on this?

**Comment by rubenromao** [comment]
Any updates??

**Comment by mautz-et-tong** [comment]
Come on @AntonEvers this is ridiculous

**Comment by rubenromao-absolute** [comment]
Same here.

**Comment by rubenromao** [comment]
Really ?!

This is more than ridiculous... 

The problem persists in magento 2.4.6-p4:

```  Problem 1
- magento/product-community-edition is locked to version 2.4.6-p4 and an update of this package was not requested.
- roave/security-advisories dev-master conflicts with tinymce/tinymce <5.10.9|>=6,<6.7.3 (magento/magento2-base 2.4.6-p4 replaces tinymce/tinymce 3.4.7).
- magento/product-community-edition 2.4.6-p4 requires magento/magento2-base 2.4.6-p4 -> satisfiable by magento/magento2-base[2.4.6-p4].
- Root composer.json requires roave/security-advisories dev-master -> satisfiable by roave/security-advisories[dev-master].```
    
Almost 3 years to fix this?

Am I missing something here?

**Comment by rrodrigonuez-i4** [comment]
Indeed, it is ridiculous...
I had to create a fork of `roave/security-advisories` and adapt it for Magento projects... That's the only way we found to keep using roave/security-advisories and not blocking the development.

**Comment by mautz-et-tong** [comment]
They simply stopped caring a long time ago. They have given up. Better switch to https://mage-os.org/

**Comment by engcom-Bravo** [comment]
Hi @t-heuser,

Thanks for your reporting and collaboration.

We have tried to reproduce the issue in Latest 2.4-develop instance and the issue is still reproducible.Kindly refer the screenshots.


<img width="1420" height="301" alt="Image" src="https://github.com/user-attachments/assets/3516d4c2-60dc-4030-b00e-949167607c50" />


Hence Confirming the issue.

Thanks.

**Comment by hostep** [comment]
The thread with comments that starts with https://github.com/magento/magento2/pull/36543#issuecomment-2003866337 may be useful in scope of this issue here

**Comment by lbajsarowicz** [comment]
composer.json on 2.4-develop has no tinymce/jquery legacy entries left. PR #36543 merged 2024-09-12 removed them, but engcom-Bravo reconfirmed this reproducible on 2025-08-08. That reconfirmation may have run against a release tag rather than 2.4-develop HEAD. Please re-QA specifically against current 2.4-develop HEAD to confirm whether this is still open.
